docs/v8-migration-api.md

```diff
@@ -0,0 +1,392 @@
+  // Babel 7
+  // [
+  //  Token (#) { value: "#", start: 10, end: 11 },
+  //  Token (name) { value: "priv", start: 11, end: 15 }
+  // ]
+  // Babel 8
+  // [
+  //  Token (privateName) { value: "priv", start: 10, end: 15 }
+  // ]
+  ```
+  **Migration**: Adapt to the new `privateName` token. If you want to restore to the Babel 7 behaviour, manually process the `privateName` token into the `#` token and the `name` token ([example](https://github.com/babel/babel/blob/7e60a93897f9a134506251ea51269faf4d02a86c/packages/babel-parser/src/parser/statement.ts#L86-L110)).
+
+- Tokenize string template as `templateNonTail` and `templateTail` ([#13919](https://github.com/babel/babel/pull/13919))
+
+  This change will affect your integration only when you are using `tokens: true` and are depending on the extra `tokens` AST output.
+
+  ```js title="babel-integration.js"
+  import { parse } from "@babel/parser";
+
+  const { tokens } = parse("`head${x}middle${y}tail`", { tokens: true });
+  console.log(tokens); // print tokens
+  // Babel 7
+  // [
+  //  Token (`),
+  //  Token (template) { value: "head" }, Token (${),
+  //  Token (name) { value: "x" }, Token (}),
+  //  Token (template) { value: "middle" }, Token (${),
+  //  Token (name) { value: "y" }, Token (}),
+  //  Token (template) { value: "tail" }
+  //  Token (`)
+  // ]
+  // Babel 8
+  // [
+  //  Token (templateNonTail) { value: "head" },
+  //  Token (name) { value: "x" },
+  //  Token (templateNonTail) { value: "middle" },
+  //  Token (name) { value: "y" },
+  //  Token (templateTail) { value: "tail" }
+  // ]
+  ```
+
+  **Migration**: Adapt to the new token design. If you want to restore to the Babel 7 behaviour, manually transform them to the Babel 7 tokens ([example](https://github.com/babel/babel/blob/7e60a93897f9a134506251ea51269faf4d02a86c/packages/babel-parser/src/parser/statement.ts#L116-L188)).
+
+### `@babel/traverse`
+
+![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+- [Allow skipped `NodePath`s to be requeued](https://github.com/babel/babel/blob/43b623c1f1e86e6fb86cae8d955a84fd924380a4/packages/babel-traverse/src/path/context.js#L241-L247) ([#13291](https://github.com/babel/babel/pull/13291))
+
+  **Notes**: `NodePath#requeue()` can requeue a skipped NodePath. This is actually a bugfix, but it causes an infinite loop in the tdz implementation of `@babel/plugin-transform-block-scoping` so we defer the change to Babel 8.
+
+![low](https://img.shields.io/badge/risk%20of%20breakage%3F-low-yellowgreen.svg)
+
+- Remove `block` argument from `Scope#rename` ([#15288](https://github.com/babel/babel/pull/15288))
+
+  ```diff
+  - rename(oldName: string, newName?: string, block?: t.Pattern | t.Scopable)
+  + rename(oldName: string, newName?: string)
+  ```
+
+  __Migration__: The third argument `block` is not used by the method. You can safely remove it if you are depending on `Scope#rename`.
```

Comment:
The third argument `block` could cause the renamer to traverse any given `block`:

https://github.com/babel/babel/blob/3d47fb696fa067799f9d25a6863a1b24f256fdea/packages/babel-traverse/src/scope/lib/renamer.ts#L153-L157

so for those passing the `block` argument other than `scope.block`, it could be a breaking change.